### PR TITLE
Money achievements

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary

SUMMARY: Content "Money achievements"

#### Purpose of change

To add more achievements

#### Describe the solution

Added two new, money related achievements:
- [x] Croesus
- [x] Allowance

#### Describe alternatives you've considered

N/A

#### Testing

Spawning 1000 cash cards in game and building a house of cards.

#### Additional context

N/A

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/your-ideas-for-new-achievements-and-conducts/24008/35

File: src/iexamine.cpp [hunk 1]
```diff
@@ -557,6 +557,9 @@ class atm_menu
 
             add_msg( m_info, "amount: %d", amount );
             u.use_charges( itype_cash_card, amount );
+            if( u.cash < 100000000 && ( u.cash + amount ) >= 100000000 ) {
+                g->events().send<event_type::becomes_millionaire>( u.getID() );
+            }
```
Comment (jbytheway) [inline src/iexamine.cpp:562]: The events you've added are very niche.  I feel like this one should e.g. be a generic 'deposit' event, which includes the amount deposited and the new balance.  I think deposits are rare enough that shouldn't overwhelm the log.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -602,6 +605,9 @@ class atm_menu
                 int to_insert = std::min( max_cap, remaining );
                 // insert whatever there's room for + the old balance.
                 cc->ammo_set( cc->ammo_default(), to_insert + cc->ammo_remaining() );
+                if( cc->ammo_remaining() == cc->ammo_capacity( money ) ) {
+                    g->events().send<event_type::fills_cashcard>( u.getID() );
+                }
```
Comment (jbytheway) [inline src/iexamine.cpp:610]: Similarly this could be a 'transfer_money' event with the new balance.

File: src/iexamine.cpp (hunk 1 above)
Comment (nexusmrsep) [reply src/iexamine.cpp:562]: What you suggest here is future-proofing, even though I can't imagine any specific use for it offhandedly, so I'll do it.

**Pull request closed without merging**